core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexUpgradeService.java

```diff
@@ -239,4 +261,67 @@ private IndexMetaData addDefaultUnitsIfNeeded(IndexMetaData indexMetaData) {
         // No changes:
         return indexMetaData;
     }
+
+
+    /**
+     * Checks the mappings of the mappings for compatibility with the current version
+     */
+    private IndexMetaData upgradeMappings(IndexMetaData indexMetaData) {
```

Comment:
The `upgradeMappings()` doesn't seem to modify the provided `indexMetaData ` variable, so I think this method should have `void`as return type?